fidelius.py

```diff
@@ -0,0 +1,203 @@
+from fastecdsa.point import Point
+
+
+# Java's Bouncycastle version of the curve 25519
+#   Only difference between BC and fastcecdsa is the value of curve.gy
+#   See https://github.com/bcgit/bc-java/blob/main/core/src/main/java/org/bouncycastle/crypto/ec/CustomNamedCurves.java#L99
+#   See the same file in below commit to find the hex decoded value of gx and gy
+#   https://github.com/bcgit/bc-java/commit/baefa5e3417c83299aa91be528aed6c23e23474c
+BC25519 = curve.Curve(
+    'BC25519',
+    curve.W25519.p,
+    curve.W25519.a,
+    curve.W25519.b,
+    curve.W25519.q,
+    curve.W25519.gx,
+    0x20ae19a1b8a086b4e01edd2c7748d14c923d4d7e6d7c61b229e9c5a27eced3d9,
+    # This is an OID specific to this implementation
+    #   taken as the next one from fastecdsa OIDs
+    b'\x01\x03\x06\x01\x04\x01\x97U\x05\x01'
+)
+
+
+class KeyMaterial:
+    def __init__(self, private_key, public_key, x509_public_key, nonce):
+        self.private_key = private_key
+        self.public_key = public_key
+        self.x509_public_key = x509_public_key
+        self.nonce = nonce
+
+    @classmethod
+    def generate(cls):
+        private_key, public_key = keys.gen_keypair(BC25519)
+        return cls._encode(private_key, public_key)
+
+    @classmethod
+    def generate_for_private_key(cls, private_key: int):
+        public_key = keys.get_public_key(private_key, BC25519)
+        return cls._encode(private_key, public_key)
+
+    @classmethod
+    def _encode(cls, private_key: int, public_key: Point):
+        private_key_base64 = cls.encode_private_key_to_base64(private_key)
+        public_key_base64 = cls.encode_public_key_to_base64(public_key)
+        x509_public_key_base64 = cls.encode_x509_public_key_to_base64(public_key)
+        nonce_base64 = cls.generate_base64_nonce()
+        return cls(private_key_base64, public_key_base64, x509_public_key_base64, nonce_base64)
+
+    @classmethod
+    def encode_private_key_to_base64(cls, key: int):
+        return base64.b64encode(
+            key.to_bytes(
+                (key.bit_length() + 7) // 8, byteorder='big'
+            )
+        ).decode('utf-8')
+
+    @classmethod
+    def encode_public_key_to_base64(cls, key: Point):
+        x_bytes = key.x.to_bytes((key.x.bit_length() + 7) // 8, byteorder='big')
+        y_bytes = key.y.to_bytes((key.y.bit_length() + 7) // 8, byteorder='big')
+        return base64.b64encode(
+            # 04 indicates uncompressed form
+            b'\x04' + x_bytes + y_bytes
+        ).decode('utf-8')
+
+    @classmethod
+    def encode_x509_public_key_to_base64(cls, key: Point):
+        # Todo: to be made compatible with java project
+        return encoding.pem.PEMEncoder().encode_public_key(key).encode()
+
+    @classmethod
+    def generate_base64_nonce(cls):
+        nonce = os.urandom(32)
+        return base64.b64encode(nonce).decode('utf-8')
+
+
+class CryptoController:
+
+    @classmethod
+    def encrypt(cls, encryption_request):
```

Comment:
What about using a dataclass instead of a dict for this?

Reply:
Updated.